inc/dashboard/namespace.php

```diff
@@ -0,0 +1,436 @@
+	$ranges = [
+		[
+			'days' => 7,
+			'label' => __( '7 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 30,
+			'label' => __( '30 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 90,
+			'label' => __( '90 days', 'altis-analytics' ),
+		],
+	];
+
+	$days = get_days_view();
+	?>
+	<div class="xb-analytics-date-range">
+		<?php
+		foreach ( $ranges as $current ) {
+			// Compare the current days value with the current days value pulled from the url query variables.
+			if ( $days === $current['days'] ) {
+				$selected = 'selected';
+			} else {
+				$selected = '';
+			}
+			?>
+			<a href="<?php echo esc_url_raw( add_query_arg( [ 'days' => $current['days'] ] ) ); ?>" class="<?php echo esc_html( "date-range-button $selected" ); ?>"><?php echo esc_html( $current['label'] ); ?></a>
+		<?php } ?>
+	</div>
+	<?php
+}
+
+/**
+ * Render the block name column data.
+ */
+function render_block_column() {
+	global $post;
+	?>
+	<strong><a href="<?php echo esc_url_raw( admin_url( '/admin.php?page=xb-analytics&post=' . $post->ID ) ); ?>"><?php echo esc_attr( $post->post_title ); ?></a></strong>
+	<?php
+}
+
+/**
+ * Render the last modified / author column data.
+ */
+function render_last_modified_author() {
+	global $post;
+	?>
+	<div class="post--last-modified">
+		<span class="post--last-modified-date"><?php echo esc_attr( get_the_modified_date( '', $post ) ); ?></span>
+		<span class="post--last-modified-time"><?php echo esc_attr( get_the_modified_time( '', $post ) ); ?></span>
+	</div>
+	<div class="post--author">
+		<?php echo esc_attr( get_the_author_meta( 'display_name', $post->post_author ) ); ?>
+	</div>
+	<?php
+}
+
+/**
+ * Render the views column data.
+ */
+function render_views() {
+	global $post;
+	$views = get_block_views( $post->post_name, [ 'days' => get_days_view() ] )['views'];
+	?>
+	<div class="post--views"><?php echo number_format_i18n( $views ); ?></div>
+	<?php
+}
+
+/**
+ * Render the average conversion rate column data.
+ */
+function render_average_conversion_rate() {
+	global $post;
+	$block = get_block_views( $post->post_name, [ 'days' => get_days_view() ] );
+	$conversions = $block['conversions'];
+	$views = $block['views'];
+	$rate = round( calculate_average_conversion_rate( [
+		'conversions' => $conversions,
+		'views' => $views,
+	] ) * 100 );
+	?>
+	<div class="post--avg-conversion-rate"><?php echo absint( $rate ); ?>%
+	<?php
+}
+
+/**
+ * Return the days url query string.
+ *
+ * @return int The number of days to show insights for. Defaults to 7.
+ */
+function get_days_view() : int {
+	return isset( $_GET['days'] ) ? wp_unslash( absint( $_GET['days'] ) ) : 7;
+}
+
+/**
+ * Calculate average conversion rate.
+ *
+ * @param array $block_data An array of block data.
+ * @param string $block_id A single block clientID.
+ *
+ * @return float The conversion rate (calculated by conversions / views).
+ */
+function calculate_average_conversion_rate( array $block_data = [], string $block_id = '' ) : float {
```

Comment:
In every instance this function has the `$block_data` passed into it and never the `$block_id`, seems it could be simplified to just accept views and conversions as arguments instead.